test/acceptance/tests/terminating-gateway/terminating_gateway_test.go

```diff
@@ -0,0 +1,94 @@
+package connect
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that terminating gateways work in a default installation.
+func TestTerminatingGateway(t *testing.T) {
+	testName := "Terminating GW non-acl"
+	t.Run(testName, func(t *testing.T) {
+		env := suite.Environment()
+
+		helmValues := map[string]string{
+			"connectInject.enabled":                    "true",
+			"terminatingGateways.enabled":              "true",
+			"terminatingGateways.gateways[0].name":     "terminating-gateway",
+			"terminatingGateways.gateways[0].replicas": "1",
+		}
+
+		t.Log("creating consul cluster")
+		releaseName := helpers.RandomName()
+		consulCluster := framework.NewHelmCluster(t, helmValues, env.DefaultContext(t), suite.Config(), releaseName)
+		consulCluster.Create(t)
+
+		// Once the cluster is up register the external service, then create the config entry.
+		consulClient := consulCluster.SetupConsulClient(t, false)
+
+		// Register the external service
+		t.Log("registering the service")
+		if err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+			Address:           "example.com",
+			Meta:              map[string]string{"external-node": "true", "external-probe": "true"},
+			ID:                "example",
+			Name:              "example-http",
+			Port:              80,
+			TaggedAddresses:   nil,
+			EnableTagOverride: false,
+		}); err != nil {
+			require.NoError(t, err)
+		}
+
+		// Create the config entry for the terminating gateway
+		t.Log("creating config entry")
+		created, _, err := consulClient.ConfigEntries().Set(&api.TerminatingGatewayConfigEntry{
+			Kind:     api.TerminatingGateway,
+			Name:     "terminating-gateway",
+			Services: []api.LinkedService{{Name: "example-http"}},
+		}, nil)
+		require.NoError(t, err)
+		require.Equal(t, true, created, "config entry failed")
+
+		k8sClient := env.DefaultContext(t).KubernetesClient(t)
+		k8sOptions := env.DefaultContext(t).KubectlOptions()
+
+		// Deploy the static client
+		t.Log("deploying static client")
+		deployStaticClient(t, suite.Config(), env.DefaultContext(t).KubectlOptions())
+
+		// Test that we can make a call to the terminating gateway
+		t.Log("trying calls to terminating gateway")
+		checkConnection(t, k8sOptions, k8sClient)
+	})
+}
+
+// checkConnection checks if static-client can connect through the terminating gateway.
+func checkConnection(t *testing.T, options *k8s.KubectlOptions, client kubernetes.Interface) {
```

Comment:
@ishustava should we make this a helper? I see the same method in every test and they're diverging. Some use the curl options `-vvvs`, others `-vvvsSf`, `-vvvsSs`.

Reply:
yeah, I think we should! I'm Ok with making this refactor outside of this PR though.